sdk/mixedreality/azure-mixedreality-authentication/src/main/java/com/azure/mixedreality/authentication/MixedRealityStsClientBuilder.java

```diff
@@ -0,0 +1,402 @@
+     * called a new instance of {@link MixedRealityStsClient} is created.
+     *
+     * @return A {@link MixedRealityStsClient} with the options set from the builder.
+     */
+    public MixedRealityStsClient buildClient() {
+        return new MixedRealityStsClient(this.buildAsyncClient());
+    }
+
+    /**
+     * Create a {@link MixedRealityStsAsyncClient} based on options set in the builder. Every time {@code buildAsyncClient()} is
+     * called a new instance of {@link MixedRealityStsAsyncClient} is created.
+     *
+     * @return A {@link MixedRealityStsAsyncClient} with the options set from the builder.
+     * @throws NullPointerException If any required values are null.
+     * @throws IllegalArgumentException If the accountId or endpoint are not properly formatted.
+     */
+    public MixedRealityStsAsyncClient buildAsyncClient() {
+        Objects.requireNonNull(this.accountId, "The 'accountId' has not been set and is required.");
+        Objects.requireNonNull(this.accountDomain, "The 'accountDomain' has not been set and is required.");
+
+        UUID accountId;
+        try {
+            accountId = UUID.fromString(this.accountId);
+        } catch (IllegalArgumentException ex) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("The 'accountId' must be a UUID formatted value."));
+        }
+
+        String endpoint;
+        if (this.endpoint != null) {
+            try {
+                new URL(this.endpoint);
+                endpoint = this.endpoint;
+            } catch (MalformedURLException ex) {
+                throw logger.logExceptionAsWarning(new IllegalArgumentException("The 'endpoint' must be a valid URL."));
+            }
+        } else {
+            endpoint = AuthenticationEndpoint.constructFromDomain(this.accountDomain);
+        }
+
+        if (this.pipeline == null) {
+            if (this.tokenCredential == null && this.keyCredential != null) {
+                this.tokenCredential = new MixedRealityAccountKeyCredential(accountId, this.keyCredential);
+            }
+
+            Objects.requireNonNull(this.tokenCredential, "The 'credential' has not been set and is required.");
+            String scope = AuthenticationEndpoint.constructScope(endpoint);
+            HttpPipelinePolicy authPolicy = new BearerTokenAuthenticationPolicy(this.tokenCredential, scope);
+            this.pipeline = createHttpPipeline(this.httpClient, authPolicy, this.customPolicies);
+        }
+
+        // MixedRealityStsServiceVersion version;
+
+        // if (this.apiVersion != null) {
+        //     version = this.apiVersion;
+        // } else {
+        //     version = MixedRealityStsServiceVersion.getLatest();
+        // }
+
+        MixedRealityStsRestClientImpl serviceClient = new MixedRealityStsRestClientImplBuilder()
+            //.apiVersion(version.getVersion())
```

Comment:
The generated code doesn't appear to let you specify the API version to use. Can anyone tell me how to fix that?

Reply:
The newer version of autorest allows you to set that.

Reply:
Which version?

Reply:
Version 4.0.16 should have this.

Reply:
Thanks. That looks better 👍 .